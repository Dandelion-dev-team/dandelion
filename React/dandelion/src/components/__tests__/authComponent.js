import React from "react"
import renderer from "react-test-renderer"
import AuthComponent from "../authComponent"


describe("Auth", () => {
    let originalFetch;

    beforeEach(() => {
        originalFetch = global.fetch;
        global.fetch = jest.fn(() => Promise.resolve({
            json: () => Promise.resolve({
                authorities: [
                    {
                        "id": "FLK",
                        "name": "Stirlingshire",
                        "telephone": "07464787634",
                        "email": "stirling@authority.com"
                    },
                    {
                        "id": "GLA",
                        "name": "Glasgow",
                        "telephone": "07424387634",
                        "email": "glasgow@authority.com"
                    }
                ]
            })
        }));
    });

    afterEach(() => {
        global.fetch = originalFetch;
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(<AuthComponent/>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})