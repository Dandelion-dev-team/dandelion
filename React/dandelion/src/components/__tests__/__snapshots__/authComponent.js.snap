// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth renders correctly 1`] = `
<div
  className="authTable"
>
  <table
    className="tableList"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Name
        </th>
        <th>
          Telephone
        </th>
        <th>
          Email
        </th>
        <th>
          Edit
        </th>
        <th>
          Delete
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Auth with data renders correctly 1`] = `
<div
  className="authTable"
>
  <table
    className="tableList"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Name
        </th>
        <th>
          Telephone
        </th>
        <th>
          Email
        </th>
        <th>
          Edit
        </th>
        <th>
          Delete
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
